packages/o/opencascade/xmake.lua

```diff
@@ -0,0 +1,632 @@
+        ["CSF_fontconfig"]    = package:is_plat("linux")     and { deps = {"fontconfig"} } or nil,
+        ["CSF_XwLibs"]        = package:is_plat("linux")     and { deps = {"libx11"} } or nil, -- ? not sure
+        -- optional deps
+        ["CSF_OpenGlLibs"]    = package:config("opengl")     and { deps = {"opengl"} } or nil,
+        -- missing opengles. TODO
+        ["CSF_TclTkLibs"]     = package:config("tk")         and { deps = {"tk"} } or nil,
+        ["CSF_FFmpeg"]        = package:config("ffmpeg")     and { deps = {"ffmpeg"} } or nil,
+        ["CSF_FreeImagePlus"] = package:config("freeimage")  and { deps = {"freeimage"} } or nil,
+        ["CSF_OpenVR"]        = package:config("openvr")     and { deps = {"openvr"} } or nil,
+        ["CSF_RapidJSON"]     = package:config("rapidjson")  and { deps = {"rapidjson"} } or nil,
+        ["CSF_Draco"]         = package:config("draco")      and { deps = {"draco"} } or nil,
+        ["CSF_TBB"]           = package:config("tbb")        and { deps = {"tbb"} } or nil,
+        ["CSF_VTK"]           = package:config("vtk")        and { deps = {"vtk"} } or nil,
+        ["CSF_MMGR"]          = nil, -- TODO for memory manager support in tkernel
+
+        -- system libs
+        ["CSF_androidlog"]    = package:is_plat("android")   and { syslinks = {"log"} } or nil,
+        ["CSF_ThreadLibs"]    = package:is_plat("linux")     and { syslinks = {"pthread", "rt"} } or nil,
+        ["CSF_dl"]            = package:is_plat("linux")     and { syslinks = {"dl"} } or nil,
+        ["CSF_dpsLibs"]       = nil,
+        ["CSF_XmuLibs"]       = nil,
+        ["CSF_advapi32"]      = package:is_plat("windows")   and { syslinks = {"advapi32"} } or nil,
+        ["CSF_gdi32"]         = package:is_plat("windows")   and { syslinks = {"gdi32"} } or nil,
+        ["CSF_psapi"]         = package:is_plat("windows")   and { syslinks = {"psapi"} } or nil,
+        ["CSF_shell32"]       = package:is_plat("windows")   and { syslinks = {"shell32"} } or nil,
+        ["CSF_user32"]        = package:is_plat("windows")   and { syslinks = {"user32"} } or nil,
+        ["CSF_winmm"]         = package:is_plat("windows")   and { syslinks = {"winmm"} } or nil,
+        ["CSF_wsock32"]       = package:is_plat("windows")   and { syslinks = {"wsock32"} } or nil,
+        ["CSF_d3d9"]          = package:config("d3d9")       and { syslinks = {"d3d9"} } or nil,
+
+        -- macOS / iOS frameworks
+        ["CSF_Appkit"]        = package:is_plat("iphoneos") and { frameworks = {"UIKit"} }
+                            or (package:is_plat("macosx") and { frameworks = {"AppKit"} } or nil),
+        ["CSF_IOKit"]         = package:is_plat("macosx")    and { frameworks = {"IOKit"} } or nil,
+        ["CSF_objc"]          = package:is_plat("macosx", "iphoneos")   and { syslinks = {"objc"} } or nil,
+        }
+
+        for cmake_dep, xmake_dep in pairs(occt_cmake_to_xmake_deps) do
```

Comment:
The order of traversal using pairs is random. This may cause the order of syslinks/frameworks to be random as well.